metadata-integration/java/datahub-client/src/main/java/datahub/client/file/FileEmitter.java

```diff
@@ -0,0 +1,186 @@
+package datahub.client.file;
+
+import java.io.BufferedWriter;
+import java.io.FileWriter;
+import java.io.IOException;
+import java.util.List;
+import java.util.concurrent.ExecutionException;
+import java.util.concurrent.Future;
+import java.util.concurrent.TimeUnit;
+import java.util.concurrent.TimeoutException;
+
+import com.fasterxml.jackson.annotation.JsonInclude;
+import com.fasterxml.jackson.core.util.DefaultIndenter;
+import com.fasterxml.jackson.core.util.DefaultPrettyPrinter;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.linkedin.data.template.JacksonDataTemplateCodec;
+import com.linkedin.mxe.MetadataChangeProposal;
+
+import datahub.client.Callback;
+import datahub.client.Emitter;
+import datahub.client.MetadataWriteResponse;
+import datahub.event.EventFormatter;
+import datahub.event.MetadataChangeProposalWrapper;
+import datahub.event.UpsertAspectRequest;
+import lombok.extern.slf4j.Slf4j;
+
+@Slf4j
+public class FileEmitter implements Emitter {
+
+  private final EventFormatter eventFormatter;
+  private final FileEmitterConfig config;
+  private final ObjectMapper objectMapper = new ObjectMapper()
+      .setSerializationInclusion(JsonInclude.Include.NON_NULL);
+  private final JacksonDataTemplateCodec dataTemplateCodec = new JacksonDataTemplateCodec(objectMapper.getFactory());
+
+  private BufferedWriter writter;
+  private boolean wroteSomething;
+
+  /**
+   * The default constructor
+   * 
+   * @param config
+   */
+  public FileEmitter(FileEmitterConfig config) {
+
+    this.config = config;
+    this.eventFormatter = this.config.getEventFormatter();
+
+    DefaultPrettyPrinter pp = new DefaultPrettyPrinter()
+        .withObjectIndenter(new DefaultIndenter("    ", DefaultIndenter.SYS_LF))
+        .withArrayIndenter(new DefaultIndenter("    ", DefaultIndenter.SYS_LF));
```

Comment:
Would it be better to define private static constant for the string with spaces/tabs?

Reply:
Done